libs/components/validation/src/lib/modules/validation/validation.ts

```diff
@@ -8,8 +12,19 @@ export abstract class SkyValidation {
     return regex.test(emailAddress);
   }
 
-  public static isUrl(url: string): boolean {
+  public static isUrl(url: string, options?: SkyUrlValidationOptions): boolean {
     const regex = /^((http|https):\/\/)?([\w-]+\.)+[\w-]+/i;
-    return regex.test(url);
+    url = url.toString();
```

Comment:
I was getting a compile error in some cases claiming that the `.test(_)` function only accepts `string`s, and adding that line fixed that error for me.

Reply:
I see what's going on. The `SkyUrlValidationDirective` is calling `SkyValidation.isUrl` with the value of its form control. The value of a form control is `any`, so if the consumer sets the value to anything other than a `string`, the `isUrl` check fails (because it's expecting a `string`). 

To make the `isUrl` method more robust (and to better explain the scenario when the `url` value is not a string), I would change the first parameter to a type of `unknown` because it's type really is unknown, coming from a form control.
```
public static isUrl(
  value: unknown,
  options?: SkyUrlValidationOptions
): boolean {
  const regex = /^((http|https):\/\/)?([\w-]+\.)+[\w-]+/i;

  if (typeof value !== 'string') {
    return false;
  }

  const url: string = value;

  if (options) {
...
```

Reply:
resolved in: [make incoming value typed as unknown and then check](https://github.com/blackbaud/skyux/pull/201/commits/b5eac3a25445552abb8040b7e159e84f74067f58)